build broken on RHEL5 because of -Werror=declaration-after-statement

...
compiling src/main.o
cc1: error: unrecognized command line option "-Werror=declaration-after-statement"
make: **\* [src/main.o] Error 1
...

[jdv@jdv rakudo]$ cat /etc/redhat-release 
CentOS release 5.11 (Final)
[jdv@jdv rakudo]$

[jdv@jdv rakudo]$ cc -v
Using built-in specs.
Target: x86_64-redhat-linux
Configured with: ../configure --prefix=/usr --mandir=/usr/share/man
--infodir=/usr/share/info --enable-shared --enable-threads=posix
--enable-checking=release --with-system-zlib --enable-__cxa_atexit
--disable-libunwind-exceptions --enable-libgcj-multifile
--enable-languages=c,c++,objc,obj-c++,java,fortran,ada --enable-java-awt=gtk
--disable-dssi --disable-plugin
--with-java-home=/usr/lib/jvm/java-1.4.2-gcj-1.4.2.0/jre --with-cpu=generic
--host=x86_64-redhat-linux
Thread model: posix
gcc version 4.1.2 20080704 (Red Hat 4.1.2-55)
[jdv@jdv rakudo]$

...
15:11 < FROGGS> yes, we added these -Werror options just recently
15:11 < jdv79> that would be my guess
15:12 < FROGGS> it is meant to force devs to care about MSVC
15:12 < FROGGS> and actually not break things :/
15:14 < FROGGS> jdv79: we have a build/probe.pm that tries to figure out what works and what not... seems that this -Werror option needs to go in there
15:14 < FROGGS> :o)
15:14 < jdv79> ok
15:14 < jdv79> neato
15:14 < FROGGS> jdv79: so, if you don't want to tackle it yourself, please open an issue in the moarvm repo
15:14 < jdv79> looks like RHEL 5 is "not EOL'd til 2017 or 2020
15:14 < jdv79> so maybe worth supporting
15:15 < FROGGS> aye
15:15 < jdv79> i'll ticket it
...

@timo, I am sorry to say I don't even know if that applies to my issue.

Here is my problem. I am trying to install some perl modules with cpan, cpan doing its magic calls GCC with some options. it one of this options that causing this my problem. 

Here is the error:
```
gcc -c  -I./Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-   after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -Werror=declaration-after-statement -Wpointer-sign -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.89\" -DXS_VERSION=\"2.89\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   Encode.c
**cc1: error: unrecognized command line option "-Werror=declaration-after-statement"**
make: *** [Encode.o] Error 1
  /usr/bin/make  -- NOT OK
```

So since in cpam download the required files before compiling i am not sure where or how to correct this error. Any ideas
 
**This is the complete stdout of the command**
 ```
cpan HTTP::Status
CPAN: Storable loaded ok
Going to read /root/.cpan/Metadata
  Database was generated on Wed, 03 May 2017 12:17:02 GMT
Running install for module HTTP::Status
Running make for E/ET/ETHER/HTTP-Message-6.11.tar.gz
CPAN: Digest::MD5 loaded ok
CPAN: Compress::Zlib loaded ok
Checksum for /root/.cpan/sources/authors/id/E/ET/ETHER/HTTP-Message-6.11.tar.gz ok
Scanning cache /root/.cpan/build for sizes
Deleting from cache: /root/.cpan/build/Encode-2.89 (46.7>10.0 MB)
HTTP-Message-6.11/
HTTP-Message-6.11/Changes
HTTP-Message-6.11/LICENSE
HTTP-Message-6.11/MANIFEST
HTTP-Message-6.11/MANIFEST.SKIP
HTTP-Message-6.11/META.json
HTTP-Message-6.11/META.yml
HTTP-Message-6.11/Makefile.PL
HTTP-Message-6.11/README
HTTP-Message-6.11/lib/
HTTP-Message-6.11/t/
HTTP-Message-6.11/t/common-req.t
HTTP-Message-6.11/t/distmanifest.t
HTTP-Message-6.11/t/headers-auth.t
HTTP-Message-6.11/t/headers-etag.t
HTTP-Message-6.11/t/headers-util.t
HTTP-Message-6.11/t/headers.t
HTTP-Message-6.11/t/http-config.t
HTTP-Message-6.11/t/message-charset.t
HTTP-Message-6.11/t/message-decode-xml.t
HTTP-Message-6.11/t/message-old.t
HTTP-Message-6.11/t/message-parts.t
HTTP-Message-6.11/t/message.t
HTTP-Message-6.11/t/request.t
HTTP-Message-6.11/t/request_type_with_data.t
HTTP-Message-6.11/t/response.t
HTTP-Message-6.11/t/status-old.t
HTTP-Message-6.11/t/status.t
HTTP-Message-6.11/lib/HTTP/
HTTP-Message-6.11/lib/HTTP/Config.pm
HTTP-Message-6.11/lib/HTTP/Headers/
HTTP-Message-6.11/lib/HTTP/Headers.pm
HTTP-Message-6.11/lib/HTTP/Message.pm
HTTP-Message-6.11/lib/HTTP/Request/
HTTP-Message-6.11/lib/HTTP/Request.pm
HTTP-Message-6.11/lib/HTTP/Response.pm
HTTP-Message-6.11/lib/HTTP/Status.pm
HTTP-Message-6.11/lib/HTTP/Request/Common.pm
HTTP-Message-6.11/lib/HTTP/Headers/Auth.pm
HTTP-Message-6.11/lib/HTTP/Headers/ETag.pm
/bin/tar: Read 7168 bytes from -
HTTP-Message-6.11/lib/HTTP/Headers/Util.pm
Removing previously used /root/.cpan/build/HTTP-Message-6.11

  CPAN.pm: Going to build E/ET/ETHER/HTTP-Message-6.11.tar.gz

Checking if your kit is complete...
Looks good
Warning: prerequisite Encode 2.21 not found. We have 2.12.
Generating a Unix-style Makefile
Writing Makefile for HTTP::Message
Writing MYMETA.yml and MYMETA.json
---- Unsatisfied dependencies detected during [E/ET/ETHER/HTTP-Message-6.11.tar.gz] -----
    Encode
Shall I follow them and prepend them to the queue
of modules we are processing right now? [yes] yes
Running make test
  Delayed until after prerequisites
Running make install
  Delayed until after prerequisites
Running install for module Encode
Running make for D/DA/DANKOGAI/Encode-2.89.tar.gz
Checksum for /root/.cpan/sources/authors/id/D/DA/DANKOGAI/Encode-2.89.tar.gz ok
Encode-2.89/
Encode-2.89/AUTHORS
Encode-2.89/bin/
Encode-2.89/Byte/
Encode-2.89/Changes
Encode-2.89/CN/
Encode-2.89/EBCDIC/
Encode-2.89/encengine.c
Encode-2.89/Encode/
Encode-2.89/Encode.pm
Encode-2.89/Encode.xs
Encode-2.89/encoding.pm
Encode-2.89/JP/
Encode-2.89/KR/
Encode-2.89/lib/
Encode-2.89/Makefile.PL
Encode-2.89/MANIFEST
Encode-2.89/META.json
Encode-2.89/META.yml
Encode-2.89/README
Encode-2.89/Symbol/
Encode-2.89/t/
Encode-2.89/TW/
Encode-2.89/ucm/
Encode-2.89/Unicode/
Encode-2.89/Unicode/Makefile.PL
Encode-2.89/Unicode/Unicode.pm
Encode-2.89/Unicode/Unicode.xs
Encode-2.89/ucm/8859-1.ucm
Encode-2.89/ucm/8859-10.ucm
Encode-2.89/ucm/8859-11.ucm
Encode-2.89/ucm/8859-13.ucm
Encode-2.89/ucm/8859-14.ucm
Encode-2.89/ucm/8859-15.ucm
Encode-2.89/ucm/8859-16.ucm
Encode-2.89/ucm/8859-2.ucm
Encode-2.89/ucm/8859-3.ucm
Encode-2.89/ucm/8859-4.ucm
Encode-2.89/ucm/8859-5.ucm
Encode-2.89/ucm/8859-6.ucm
Encode-2.89/ucm/8859-7.ucm
Encode-2.89/ucm/8859-8.ucm
Encode-2.89/ucm/8859-9.ucm
Encode-2.89/ucm/adobeStdenc.ucm
Encode-2.89/ucm/adobeSymbol.ucm
Encode-2.89/ucm/adobeZdingbat.ucm
Encode-2.89/ucm/ascii.ucm
Encode-2.89/ucm/big5-eten.ucm
Encode-2.89/ucm/big5-hkscs.ucm
Encode-2.89/ucm/cp037.ucm
Encode-2.89/ucm/cp1006.ucm
Encode-2.89/ucm/cp1026.ucm
Encode-2.89/ucm/cp1047.ucm
Encode-2.89/ucm/cp1250.ucm
Encode-2.89/ucm/cp1251.ucm
Encode-2.89/ucm/cp1252.ucm
Encode-2.89/ucm/cp1253.ucm
Encode-2.89/ucm/cp1254.ucm
Encode-2.89/ucm/cp1255.ucm
Encode-2.89/ucm/cp1256.ucm
Encode-2.89/ucm/cp1257.ucm
Encode-2.89/ucm/cp1258.ucm
Encode-2.89/ucm/cp424.ucm
Encode-2.89/ucm/cp437.ucm
Encode-2.89/ucm/cp500.ucm
Encode-2.89/ucm/cp737.ucm
Encode-2.89/ucm/cp775.ucm
Encode-2.89/ucm/cp850.ucm
Encode-2.89/ucm/cp852.ucm
Encode-2.89/ucm/cp855.ucm
Encode-2.89/ucm/cp856.ucm
Encode-2.89/ucm/cp857.ucm
Encode-2.89/ucm/cp858.ucm
Encode-2.89/ucm/cp860.ucm
Encode-2.89/ucm/cp861.ucm
Encode-2.89/ucm/cp862.ucm
Encode-2.89/ucm/cp863.ucm
Encode-2.89/ucm/cp864.ucm
Encode-2.89/ucm/cp865.ucm
Encode-2.89/ucm/cp866.ucm
Encode-2.89/ucm/cp869.ucm
Encode-2.89/ucm/cp874.ucm
Encode-2.89/ucm/cp875.ucm
Encode-2.89/ucm/cp932.ucm
Encode-2.89/ucm/cp936.ucm
Encode-2.89/ucm/cp949.ucm
Encode-2.89/ucm/cp950.ucm
Encode-2.89/ucm/ctrl.ucm
Encode-2.89/ucm/dingbats.ucm
Encode-2.89/ucm/euc-cn.ucm
Encode-2.89/ucm/euc-jp.ucm
Encode-2.89/ucm/euc-kr.ucm
Encode-2.89/ucm/gb12345.ucm
Encode-2.89/ucm/gb2312.ucm
Encode-2.89/ucm/hp-roman8.ucm
Encode-2.89/ucm/ir-165.ucm
Encode-2.89/ucm/jis0201.ucm
Encode-2.89/ucm/jis0208.ucm
Encode-2.89/ucm/jis0212.ucm
Encode-2.89/ucm/johab.ucm
Encode-2.89/ucm/koi8-f.ucm
Encode-2.89/ucm/koi8-r.ucm
Encode-2.89/ucm/koi8-u.ucm
Encode-2.89/ucm/ksc5601.ucm
Encode-2.89/ucm/macArabic.ucm
Encode-2.89/ucm/macCentEuro.ucm
Encode-2.89/ucm/macChinsimp.ucm
Encode-2.89/ucm/macChintrad.ucm
Encode-2.89/ucm/macCroatian.ucm
Encode-2.89/ucm/macCyrillic.ucm
Encode-2.89/ucm/macDingbats.ucm
Encode-2.89/ucm/macFarsi.ucm
Encode-2.89/ucm/macGreek.ucm
Encode-2.89/ucm/macHebrew.ucm
Encode-2.89/ucm/macIceland.ucm
Encode-2.89/ucm/macJapanese.ucm
Encode-2.89/ucm/macKorean.ucm
Encode-2.89/ucm/macRoman.ucm
Encode-2.89/ucm/macROMnn.ucm
Encode-2.89/ucm/macRUMnn.ucm
Encode-2.89/ucm/macSami.ucm
Encode-2.89/ucm/macSymbol.ucm
Encode-2.89/ucm/macThai.ucm
Encode-2.89/ucm/macTurkish.ucm
Encode-2.89/ucm/macUkraine.ucm
Encode-2.89/ucm/nextstep.ucm
Encode-2.89/ucm/null.ucm
Encode-2.89/ucm/posix-bc.ucm
Encode-2.89/ucm/shiftjis.ucm
Encode-2.89/ucm/symbol.ucm
Encode-2.89/ucm/viscii.ucm
Encode-2.89/TW/Makefile.PL
Encode-2.89/TW/TW.pm
Encode-2.89/t/Aliases.t
Encode-2.89/t/at-cn.t
Encode-2.89/t/at-tw.t
Encode-2.89/t/big5-eten.enc
Encode-2.89/t/big5-eten.utf
Encode-2.89/t/big5-hkscs.enc
Encode-2.89/t/big5-hkscs.utf
Encode-2.89/t/CJKT.t
Encode-2.89/t/cow.t
Encode-2.89/t/decode.t
Encode-2.89/t/enc_data.t
Encode-2.89/t/enc_eucjp.t
Encode-2.89/t/enc_module.enc
Encode-2.89/t/enc_module.t
Encode-2.89/t/enc_utf8.t
Encode-2.89/t/Encode.t
Encode-2.89/t/Encoder.t
Encode-2.89/t/encoding-locale.t
Encode-2.89/t/encoding.t
Encode-2.89/t/fallback.t
Encode-2.89/t/from_to.t
Encode-2.89/t/gb2312.enc
Encode-2.89/t/gb2312.utf
Encode-2.89/t/grow.t
Encode-2.89/t/gsm0338.t
Encode-2.89/t/guess.t
Encode-2.89/t/isa.t
Encode-2.89/t/jis7-fallback.t
Encode-2.89/t/jisx0201.enc
Encode-2.89/t/jisx0201.utf
Encode-2.89/t/jisx0208.enc
Encode-2.89/t/jisx0208.utf
Encode-2.89/t/jisx0212.enc
Encode-2.89/t/jisx0212.utf
Encode-2.89/t/jperl.t
Encode-2.89/t/ksc5601.enc
Encode-2.89/t/ksc5601.utf
Encode-2.89/t/magic.t
Encode-2.89/t/mime-header.t
Encode-2.89/t/mime-name.t
Encode-2.89/t/mime_header_iso2022jp.t
Encode-2.89/t/Mod_EUCJP.pm
Encode-2.89/t/perlio.t
Encode-2.89/t/piconv.t
Encode-2.89/t/rt.pl
Encode-2.89/t/rt113164.t
Encode-2.89/t/rt65541.t
Encode-2.89/t/rt76824.t
Encode-2.89/t/rt85489.t
Encode-2.89/t/rt86327.t
Encode-2.89/t/taint.t
Encode-2.89/t/unibench.pl
Encode-2.89/t/Unicode.t
Encode-2.89/t/utf8messages.t
Encode-2.89/t/utf8ref.t
Encode-2.89/t/utf8strict.t
Encode-2.89/t/utf8warnings.t
Encode-2.89/Symbol/Makefile.PL
Encode-2.89/Symbol/Symbol.pm
Encode-2.89/lib/Encode/
Encode-2.89/lib/Encode/Alias.pm
Encode-2.89/lib/Encode/CJKConstants.pm
Encode-2.89/lib/Encode/CN/
Encode-2.89/lib/Encode/Config.pm
Encode-2.89/lib/Encode/Encoder.pm
Encode-2.89/lib/Encode/Encoding.pm
Encode-2.89/lib/Encode/GSM0338.pm
Encode-2.89/lib/Encode/Guess.pm
Encode-2.89/lib/Encode/JP/
Encode-2.89/lib/Encode/KR/
Encode-2.89/lib/Encode/MIME/
Encode-2.89/lib/Encode/PerlIO.pod
Encode-2.89/lib/Encode/Supported.pod
Encode-2.89/lib/Encode/Unicode/
Encode-2.89/lib/Encode/Unicode/UTF7.pm
Encode-2.89/lib/Encode/MIME/Header/
Encode-2.89/lib/Encode/MIME/Header.pm
Encode-2.89/lib/Encode/MIME/Name.pm
Encode-2.89/lib/Encode/MIME/Header/ISO_2022_JP.pm
Encode-2.89/lib/Encode/KR/2022_KR.pm
Encode-2.89/lib/Encode/JP/H2Z.pm
Encode-2.89/lib/Encode/JP/JIS7.pm
Encode-2.89/lib/Encode/CN/HZ.pm
Encode-2.89/KR/KR.pm
Encode-2.89/KR/Makefile.PL
Encode-2.89/JP/JP.pm
Encode-2.89/JP/Makefile.PL
Encode-2.89/Encode/_PM.e2x
Encode-2.89/Encode/_T.e2x
Encode-2.89/Encode/Changes.e2x
Encode-2.89/Encode/ConfigLocal_PM.e2x
Encode-2.89/Encode/encode.h
Encode-2.89/Encode/Makefile_PL.e2x
Encode-2.89/Encode/README.e2x
Encode-2.89/EBCDIC/EBCDIC.pm
Encode-2.89/EBCDIC/Makefile.PL
Encode-2.89/CN/CN.pm
Encode-2.89/CN/Makefile.PL
Encode-2.89/Byte/Byte.pm
Encode-2.89/Byte/Makefile.PL
Encode-2.89/bin/enc2xs
Encode-2.89/bin/encguess
Encode-2.89/bin/piconv
Encode-2.89/bin/ucm2table
Encode-2.89/bin/ucmlint
Encode-2.89/bin/ucmsort
Encode-2.89/bin/unidump
/bin/tar: Read 2560 bytes from -

  CPAN.pm: Going to build D/DA/DANKOGAI/Encode-2.89.tar.gz

Checking if your kit is complete...
Looks good
Writing MYMETA.yml and MYMETA.json
Writing MYMETA.yml and MYMETA.json
Writing MYMETA.yml and MYMETA.json
Writing MYMETA.yml and MYMETA.json
Writing MYMETA.yml and MYMETA.json
Writing MYMETA.yml and MYMETA.json
Writing MYMETA.yml and MYMETA.json
Writing MYMETA.yml and MYMETA.json
Generating a Unix-style Makefile
Writing Makefile for Encode
Writing MYMETA.yml and MYMETA.json
cp lib/Encode/JP/H2Z.pm blib/lib/Encode/JP/H2Z.pm
cp lib/Encode/JP/JIS7.pm blib/lib/Encode/JP/JIS7.pm
cp lib/Encode/Alias.pm blib/lib/Encode/Alias.pm
cp Encode/README.e2x blib/lib/Encode/README.e2x
cp lib/Encode/Unicode/UTF7.pm blib/lib/Encode/Unicode/UTF7.pm
cp lib/Encode/MIME/Header/ISO_2022_JP.pm blib/lib/Encode/MIME/Header/ISO_2022_JP.pm
cp lib/Encode/PerlIO.pod blib/lib/Encode/PerlIO.pod
cp lib/Encode/GSM0338.pm blib/lib/Encode/GSM0338.pm
cp Encode.pm blib/lib/Encode.pm
cp Encode/Makefile_PL.e2x blib/lib/Encode/Makefile_PL.e2x
cp Encode/Changes.e2x blib/lib/Encode/Changes.e2x
cp lib/Encode/Encoding.pm blib/lib/Encode/Encoding.pm
cp Encode/ConfigLocal_PM.e2x blib/lib/Encode/ConfigLocal_PM.e2x
cp Encode/encode.h blib/lib/Encode/encode.h
cp lib/Encode/KR/2022_KR.pm blib/lib/Encode/KR/2022_KR.pm
cp lib/Encode/Guess.pm blib/lib/Encode/Guess.pm
cp encoding.pm blib/lib/encoding.pm
cp lib/Encode/MIME/Name.pm blib/lib/Encode/MIME/Name.pm
cp Encode/_T.e2x blib/lib/Encode/_T.e2x
cp lib/Encode/Supported.pod blib/lib/Encode/Supported.pod
cp lib/Encode/Config.pm blib/lib/Encode/Config.pm
cp lib/Encode/Encoder.pm blib/lib/Encode/Encoder.pm
cp lib/Encode/MIME/Header.pm blib/lib/Encode/MIME/Header.pm
cp Encode/_PM.e2x blib/lib/Encode/_PM.e2x
cp lib/Encode/CN/HZ.pm blib/lib/Encode/CN/HZ.pm
cp lib/Encode/CJKConstants.pm blib/lib/Encode/CJKConstants.pm
make[1]: Entering directory `/root/.cpan/build/Encode-2.89/Byte'
cp Byte.pm ../blib/lib/Encode/Byte.pm
Running Mkbootstrap for Byte ()
chmod 644 "Byte.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Byte.bs ../blib/arch/auto/Encode/Byte/Byte.bs 644
"/usr/bin/perl"  ../bin/enc2xs -"Q" -"O" -o byte_t.c -f byte_t.fnm
Reading iso-8859-2 (iso-8859-2)
Reading iso-8859-3 (iso-8859-3)
Reading iso-8859-4 (iso-8859-4)
Reading iso-8859-5 (iso-8859-5)
Reading iso-8859-6 (iso-8859-6)
Reading iso-8859-7 (iso-8859-7)
Reading iso-8859-8 (iso-8859-8)
Reading iso-8859-9 (iso-8859-9)
Reading iso-8859-10 (iso-8859-10)
Reading iso-8859-11 (iso-8859-11)
Reading iso-8859-13 (iso-8859-13)
Reading iso-8859-14 (iso-8859-14)
Reading iso-8859-15 (iso-8859-15)
Reading iso-8859-16 (iso-8859-16)
Reading AdobeStandardEncoding (AdobeStandardEncoding)
Reading cp1006 (cp1006)
Reading cp1250 (cp1250)
Reading cp1251 (cp1251)
Reading cp1253 (cp1253)
Reading cp1254 (cp1254)
Reading cp1255 (cp1255)
Reading cp1256 (cp1256)
Reading cp1257 (cp1257)
Reading cp1258 (cp1258)
Reading cp424 (cp424)
Reading cp437 (cp437)
Reading cp737 (cp737)
Reading cp775 (cp775)
Reading cp850 (cp850)
Reading cp852 (cp852)
Reading cp855 (cp855)
Reading cp856 (cp856)
Reading cp857 (cp857)
Reading cp858 (cp858)
Reading cp860 (cp860)
Reading cp861 (cp861)
Reading cp862 (cp862)
Reading cp863 (cp863)
Reading cp864 (cp864)
Reading cp865 (cp865)
Reading cp866 (cp866)
Reading cp869 (cp869)
Reading cp874 (cp874)
Reading hp-roman8 (hp-roman8)
Reading koi8-f (koi8-f)
Reading koi8-r (koi8-r)
Reading koi8-u (koi8-u)
Reading MacArabic (MacArabic)
Reading MacCentralEurRoman (MacCentralEurRoman)
Reading MacCroatian (MacCroatian)
Reading MacCyrillic (MacCyrillic)
Reading MacFarsi (MacFarsi)
Reading MacGreek (MacGreek)
Reading MacHebrew (MacHebrew)
Reading MacIcelandic (MacIcelandic)
Reading MacRomanian (MacRomanian)
Reading MacRumanian (MacRumanian)
Reading MacRoman (MacRoman)
Reading MacSami (MacSami)
Reading MacThai (MacThai)
Reading MacTurkish (MacTurkish)
Reading MacUkrainian (MacUkrainian)
Reading nextstep (nextstep)
Reading viscii (viscii)
"/usr/bin/perl" "/usr/lib/perl5/5.8.8/ExtUtils/xsubpp"  -nolinenumbers -typemap '/usr/lib/perl5/5.8.8/ExtUtils/typemap'  Byte.xs > Byte.xsc
mv Byte.xsc Byte.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   Byte.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   byte_t.c
rm -f ../blib/arch/auto/Encode/Byte/Byte.so
gcc  -shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic Byte.o byte_t.o  -o ../blib/arch/auto/Encode/Byte/Byte.so  \
	      \

chmod 755 ../blib/arch/auto/Encode/Byte/Byte.so
make[1]: Leaving directory `/root/.cpan/build/Encode-2.89/Byte'
make[1]: Entering directory `/root/.cpan/build/Encode-2.89/CN'
cp CN.pm ../blib/lib/Encode/CN.pm
Running Mkbootstrap for CN ()
chmod 644 "CN.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- CN.bs ../blib/arch/auto/Encode/CN/CN.bs 644
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o cp_00_t.c -f cp_00_t.fnm
Reading cp936 (cp936)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o eu_01_t.c -f eu_01_t.fnm
Reading euc-cn (euc-cn)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o gb_02_t.c -f gb_02_t.fnm
Reading gb12345-raw (gb12345-raw)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o gb_03_t.c -f gb_03_t.fnm
Reading gb2312-raw (gb2312-raw)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o ir_04_t.c -f ir_04_t.fnm
Reading iso-ir-165 (iso-ir-165)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o ma_05_t.c -f ma_05_t.fnm
Reading MacChineseSimp (MacChineseSimp)
"/usr/bin/perl" "/usr/lib/perl5/5.8.8/ExtUtils/xsubpp" -noprototypes -nolinenumbers -typemap '/usr/lib/perl5/5.8.8/ExtUtils/typemap'  CN.xs > CN.xsc
mv CN.xsc CN.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   CN.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   cp_00_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   eu_01_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   gb_02_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   gb_03_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   ir_04_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   ma_05_t.c
rm -f ../blib/arch/auto/Encode/CN/CN.so
gcc  -shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic CN.o cp_00_t.o eu_01_t.o gb_02_t.o gb_03_t.o ir_04_t.o ma_05_t.o  -o ../blib/arch/auto/Encode/CN/CN.so  \
	      \

chmod 755 ../blib/arch/auto/Encode/CN/CN.so
make[1]: Leaving directory `/root/.cpan/build/Encode-2.89/CN'
make[1]: Entering directory `/root/.cpan/build/Encode-2.89/EBCDIC'
cp EBCDIC.pm ../blib/lib/Encode/EBCDIC.pm
Running Mkbootstrap for EBCDIC ()
chmod 644 "EBCDIC.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- EBCDIC.bs ../blib/arch/auto/Encode/EBCDIC/EBCDIC.bs 644
"/usr/bin/perl"  ../bin/enc2xs -"Q" -"O" -o ebcdic_t.c -f ebcdic_t.fnm
Reading cp37 (cp37)
Reading cp1026 (cp1026)
Reading cp1047 (cp1047)
Reading cp500 (cp500)
Reading cp875 (cp875)
Reading posix-bc (posix-bc)
"/usr/bin/perl" "/usr/lib/perl5/5.8.8/ExtUtils/xsubpp"  -nolinenumbers -typemap '/usr/lib/perl5/5.8.8/ExtUtils/typemap'  EBCDIC.xs > EBCDIC.xsc
mv EBCDIC.xsc EBCDIC.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.02\" -DXS_VERSION=\"2.02\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   EBCDIC.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.02\" -DXS_VERSION=\"2.02\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   ebcdic_t.c
rm -f ../blib/arch/auto/Encode/EBCDIC/EBCDIC.so
gcc  -shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic EBCDIC.o ebcdic_t.o  -o ../blib/arch/auto/Encode/EBCDIC/EBCDIC.so  \
	      \

chmod 755 ../blib/arch/auto/Encode/EBCDIC/EBCDIC.so
make[1]: Leaving directory `/root/.cpan/build/Encode-2.89/EBCDIC'
make[1]: Entering directory `/root/.cpan/build/Encode-2.89/JP'
cp JP.pm ../blib/lib/Encode/JP.pm
Running Mkbootstrap for JP ()
chmod 644 "JP.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- JP.bs ../blib/arch/auto/Encode/JP/JP.bs 644
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o cp_00_t.c -f cp_00_t.fnm
Reading cp932 (cp932)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o eu_01_t.c -f eu_01_t.fnm
Reading euc-jp (euc-jp)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o ji_02_t.c -f ji_02_t.fnm
Reading jis0201-raw (jis0201-raw)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o ji_03_t.c -f ji_03_t.fnm
Reading jis0208-raw (jis0208-raw)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o ji_04_t.c -f ji_04_t.fnm
Reading jis0212-raw (jis0212-raw)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o ma_05_t.c -f ma_05_t.fnm
Reading MacJapanese (MacJapanese)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o sh_06_t.c -f sh_06_t.fnm
Reading shiftjis (shiftjis)
"/usr/bin/perl" "/usr/lib/perl5/5.8.8/ExtUtils/xsubpp"  -nolinenumbers -typemap '/usr/lib/perl5/5.8.8/ExtUtils/typemap'  JP.xs > JP.xsc
mv JP.xsc JP.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   JP.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   cp_00_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   eu_01_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   ji_02_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   ji_03_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   ji_04_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   ma_05_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   sh_06_t.c
rm -f ../blib/arch/auto/Encode/JP/JP.so
gcc  -shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic JP.o cp_00_t.o eu_01_t.o ji_02_t.o ji_03_t.o ji_04_t.o ma_05_t.o sh_06_t.o  -o ../blib/arch/auto/Encode/JP/JP.so  \
	      \

chmod 755 ../blib/arch/auto/Encode/JP/JP.so
make[1]: Leaving directory `/root/.cpan/build/Encode-2.89/JP'
make[1]: Entering directory `/root/.cpan/build/Encode-2.89/KR'
cp KR.pm ../blib/lib/Encode/KR.pm
Running Mkbootstrap for KR ()
chmod 644 "KR.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- KR.bs ../blib/arch/auto/Encode/KR/KR.bs 644
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o cp_00_t.c -f cp_00_t.fnm
Reading cp949 (cp949)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o eu_01_t.c -f eu_01_t.fnm
Reading euc-kr (euc-kr)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o jo_02_t.c -f jo_02_t.fnm
Reading johab (johab)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o ks_03_t.c -f ks_03_t.fnm
Reading ksc5601-raw (ksc5601-raw)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o ma_04_t.c -f ma_04_t.fnm
Reading MacKorean (MacKorean)
"/usr/bin/perl" "/usr/lib/perl5/5.8.8/ExtUtils/xsubpp"  -nolinenumbers -typemap '/usr/lib/perl5/5.8.8/ExtUtils/typemap'  KR.xs > KR.xsc
mv KR.xsc KR.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   KR.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   cp_00_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   eu_01_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   jo_02_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   ks_03_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   ma_04_t.c
rm -f ../blib/arch/auto/Encode/KR/KR.so
gcc  -shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic KR.o cp_00_t.o eu_01_t.o jo_02_t.o ks_03_t.o ma_04_t.o  -o ../blib/arch/auto/Encode/KR/KR.so  \
	      \

chmod 755 ../blib/arch/auto/Encode/KR/KR.so
make[1]: Leaving directory `/root/.cpan/build/Encode-2.89/KR'
make[1]: Entering directory `/root/.cpan/build/Encode-2.89/Symbol'
cp Symbol.pm ../blib/lib/Encode/Symbol.pm
Running Mkbootstrap for Symbol ()
chmod 644 "Symbol.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Symbol.bs ../blib/arch/auto/Encode/Symbol/Symbol.bs 644
"/usr/bin/perl"  ../bin/enc2xs -"Q" -"O" -o symbol_t.c -f symbol_t.fnm
Reading AdobeSymbol (AdobeSymbol)
Reading AdobeZdingbat (AdobeZdingbat)
Reading dingbats (dingbats)
Reading MacDingbats (MacDingbats)
Reading MacSymbol (MacSymbol)
Reading symbol (symbol)
"/usr/bin/perl" "/usr/lib/perl5/5.8.8/ExtUtils/xsubpp"  -nolinenumbers -typemap '/usr/lib/perl5/5.8.8/ExtUtils/typemap'  Symbol.xs > Symbol.xsc
mv Symbol.xsc Symbol.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.02\" -DXS_VERSION=\"2.02\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   Symbol.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.02\" -DXS_VERSION=\"2.02\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   symbol_t.c
rm -f ../blib/arch/auto/Encode/Symbol/Symbol.so
gcc  -shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic Symbol.o symbol_t.o  -o ../blib/arch/auto/Encode/Symbol/Symbol.so  \
	      \

chmod 755 ../blib/arch/auto/Encode/Symbol/Symbol.so
make[1]: Leaving directory `/root/.cpan/build/Encode-2.89/Symbol'
make[1]: Entering directory `/root/.cpan/build/Encode-2.89/TW'
cp TW.pm ../blib/lib/Encode/TW.pm
Running Mkbootstrap for TW ()
chmod 644 "TW.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- TW.bs ../blib/arch/auto/Encode/TW/TW.bs 644
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o bi_00_t.c -f bi_00_t.fnm
Reading big5-eten (big5-eten)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o bi_01_t.c -f bi_01_t.fnm
Reading big5-hkscs (big5-hkscs)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o cp_02_t.c -f cp_02_t.fnm
Reading cp950 (cp950)
"/usr/bin/perl"  ../bin/enc2xs -"Q" -o ma_03_t.c -f ma_03_t.fnm
Reading MacChineseTrad (MacChineseTrad)
"/usr/bin/perl" "/usr/lib/perl5/5.8.8/ExtUtils/xsubpp"  -nolinenumbers -typemap '/usr/lib/perl5/5.8.8/ExtUtils/typemap'  TW.xs > TW.xsc
mv TW.xsc TW.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   TW.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   bi_00_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   bi_01_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   cp_02_t.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.03\" -DXS_VERSION=\"2.03\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   ma_03_t.c
rm -f ../blib/arch/auto/Encode/TW/TW.so
gcc  -shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic TW.o bi_00_t.o bi_01_t.o cp_02_t.o ma_03_t.o  -o ../blib/arch/auto/Encode/TW/TW.so  \
	      \

chmod 755 ../blib/arch/auto/Encode/TW/TW.so
make[1]: Leaving directory `/root/.cpan/build/Encode-2.89/TW'
make[1]: Entering directory `/root/.cpan/build/Encode-2.89/Unicode'
cp Unicode.pm ../blib/lib/Encode/Unicode.pm
Running Mkbootstrap for Unicode ()
chmod 644 "Unicode.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Unicode.bs ../blib/arch/auto/Encode/Unicode/Unicode.bs 644
"/usr/bin/perl" "/usr/lib/perl5/5.8.8/ExtUtils/xsubpp"  -typemap '/usr/lib/perl5/5.8.8/ExtUtils/typemap'  Unicode.xs > Unicode.xsc
mv Unicode.xsc Unicode.c
gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.15\" -DXS_VERSION=\"2.15\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   Unicode.c
rm -f ../blib/arch/auto/Encode/Unicode/Unicode.so
gcc  -shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic Unicode.o  -o ../blib/arch/auto/Encode/Unicode/Unicode.so  \
	      \

chmod 755 ../blib/arch/auto/Encode/Unicode/Unicode.so
make[1]: Leaving directory `/root/.cpan/build/Encode-2.89/Unicode'
Running Mkbootstrap for Encode ()
chmod 644 "Encode.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Encode.bs blib/arch/auto/Encode/Encode.bs 644
"/usr/bin/perl" "/usr/lib/perl5/5.8.8/ExtUtils/xsubpp"  -typemap '/usr/lib/perl5/5.8.8/ExtUtils/typemap'  Encode.xs > Encode.xsc
mv Encode.xsc Encode.c
"/usr/bin/perl"  bin/enc2xs -"Q" -"O" -o def_t.c -f def_t.fnm
Reading iso-8859-1 (iso-8859-1)
Reading ascii (ascii)
Reading cp1252 (cp1252)
Reading ascii-ctrl (ascii-ctrl)
Reading null (null)
gcc -c  -I./Encode -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm -Werror=declaration-after-statement -Wpointer-sign -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.89\" -DXS_VERSION=\"2.89\" -fPIC "-I/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE"   Encode.c
cc1: error: unrecognized command line option "-Werror=declaration-after-statement"
make: *** [Encode.o] Error 1
  /usr/bin/make  -- NOT OK
Running make test
  Can't test without successful make
Running make install
  make had returned bad status, install seems impossible
Running make for E/ET/ETHER/HTTP-Message-6.11.tar.gz
  Is already unwrapped into directory /root/.cpan/build/HTTP-Message-6.11

  CPAN.pm: Going to build E/ET/ETHER/HTTP-Message-6.11.tar.gz

cp lib/HTTP/Status.pm blib/lib/HTTP/Status.pm
cp lib/HTTP/Headers/Auth.pm blib/lib/HTTP/Headers/Auth.pm
cp lib/HTTP/Request.pm blib/lib/HTTP/Request.pm
cp lib/HTTP/Headers/ETag.pm blib/lib/HTTP/Headers/ETag.pm
cp lib/HTTP/Request/Common.pm blib/lib/HTTP/Request/Common.pm
cp lib/HTTP/Headers/Util.pm blib/lib/HTTP/Headers/Util.pm
cp lib/HTTP/Config.pm blib/lib/HTTP/Config.pm
cp lib/HTTP/Message.pm blib/lib/HTTP/Message.pm
cp lib/HTTP/Response.pm blib/lib/HTTP/Response.pm
cp lib/HTTP/Headers.pm blib/lib/HTTP/Headers.pm
Manifying 8 pod documents
  /usr/bin/make  -- OK
Running make test
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/common-req................ok
t/distmanifest..............skipped
        all skipped: these tests are for authors only!
t/headers-auth..............ok
t/headers-etag..............ok
t/headers-util..............ok
t/headers...................ok
t/http-config...............ok
t/message-charset...........ok 1/43Can't locate object method "mime_name" via package "Encode::utf8" at /root/.cpan/build/HTTP-Message-6.11/blib/lib/HTTP/Message.pm line 251.
# Looks like your test exited with 9 just after 6.
t/message-charset...........dubious
	Test returned status 9 (wstat 2304, 0x900)
DIED. FAILED tests 7-43
	Failed 37/43 tests, 13.95% okay
t/message-decode-xml........ok
t/message-old...............ok
t/message-parts.............ok
t/message...................ok
t/request...................ok
t/request_type_with_data....ok
t/response..................ok
t/status-old................ok
t/status....................ok
Failed Test         Stat Wstat Total Fail  Failed  List of Failed
-------------------------------------------------------------------------------
t/message-charset.t    9  2304    43   74 172.09%  7-43
1 test skipped.
Failed 1/17 test scripts, 94.12% okay. 37/563 subtests failed, 93.43% okay.
make: *** [test_dynamic] Error 255
  /usr/bin/make test -- NOT OK
Running make install
  make test had returned bad status, won't install without force
```

--- a/Configure.pl
+++ b/Configure.pl
@@ -143,10 +143,24 @@ $config{translate_newline_output} = 0  unless defined $config{translate_newline_
 $config{ld}           = $config{cc} unless defined $config{ld};
 $config{ldout}        = $config{ccout} unless defined $config{ldout};
 $config{ldsys}        = $config{ldusr} unless defined $config{ldsys};
-$config{ldmiscflags}  = $config{ccmiscflags} unless defined $config{ldmiscflags};
 $config{ldoptiflags}  = $config{ccoptiflags} unless defined $config{ldoptiflags};
 $config{lddebugflags} = $config{ccdebugflags} unless defined $config{lddebugflags};
 $config{ldinstflags}  = $config{ccinstflags} unless defined $config{ldinstflags};
+
+# Probe the compiler.
+build::probe::compiler_usability(\%config, \%defaults);
+
+# Remove unsupported -Werror=* gcc flags if gcc doesn't support them.
+build::probe::specific_werror(\%config, \%defaults);
+if ($config{cc} eq 'gcc' && !$config{can_specific_werror}) {
+    $config{ccmiscflags} =~ s/-Werror=[^ ]+//g;
+    $config{ccmiscflags} =~ s/ +/ /g;
+    $config{ccmiscflags} =~ s/^ +$//;
+}
+
+# Set the remaining ldmiscflags. Do this after probing for gcc -Werror probe to not miss that change for the linker.
+$config{ldmiscflags}  = $config{ccmiscflags} unless defined $config{ldmiscflags};
+
 
 if ($args{'has-sha'}) {
     $config{shaincludedir} = '/usr/include/sha';
@@ -167,7 +181,7 @@ if (-e '3rdparty/libuv/src/unix/threadpool' . $defaults{obj}
 
 # test whether pkg-config works
 if (-e "$config{pkgconfig}") {
-    print("\nTesting pkgconfig ... ");
+    print dots("    Testing pkgconfig");
     system("$config{pkgconfig}", "--version");
     if ( $? == 0 ) {
         $config{pkgconfig_works} = 1;
@@ -380,18 +394,16 @@ $config{mainlibs} = '-lubsan ' . $config{mainlibs} if $args{ubsan};
 my @auxfiles = @{ $defaults{-auxfiles} };
 $config{auxclean} = @auxfiles ? '$(RM) ' . join ' ', @auxfiles : '@:';
 
-print "OK\n";
+print "OK\n\n";
 
 if ($config{crossconf}) {
     build::auto::detect_cross(\%config, \%defaults);
-    build::probe::compiler_usability(\%config, \%defaults);
     build::probe::static_inline_cross(\%config, \%defaults);
     build::probe::unaligned_access_cross(\%config, \%defaults);
     build::probe::ptr_size_cross(\%config, \%defaults);
 }
 else {
     build::auto::detect_native(\%config, \%defaults);
-    build::probe::compiler_usability(\%config, \%defaults);
     build::probe::static_inline_native(\%config, \%defaults);
     build::probe::unaligned_access(\%config, \%defaults);
     build::probe::ptr_size_native(\%config, \%defaults);
@@ -576,6 +588,7 @@ sub setup_native {
     my ($os) = @_;
 
     print dots("Configuring native build environment");
+    print "\n";
 
     $os = build::probe::win32_compiler_toolchain(\%config, \%defaults)
         if $os eq 'MSWin32';
@@ -749,8 +762,9 @@ sub generate {
 sub dots {
     my $message = shift;
     my $length = shift || 55;
-
-    return "$message ". '.' x ($length - length $message) . ' ';
+    my $dot_count = $length - length $message;
+    $dot_count = 0 if $dot_count < 0;
+    return "$message " . '.' x $dot_count . ' ';
 }
 
 # fail but continue

--- a/build/probe.pm
+++ b/build/probe.pm
@@ -104,7 +104,7 @@ EOT
 
     if ($can_compile) {
 	$ENV{LDFLAGS} //= '';
-        $command = "$config->{ld} $ENV{LDFLAGS} $config->{ldout}$leaf $obj $config->{ldlibs} 2>&1";
+        $command = "$config->{ld} $ENV{LDFLAGS} $config->{ldout}$leaf $obj 2>&1";
         $output  = `$command` || $!;
         if ($? >> 8 == 0) {
             $can_link = 1;
@@ -225,6 +225,34 @@ sub static_inline_cross {
     # might get confused by link time optimisations that only fail at run time,
     # which the system test does detect.
     $config->{static_inline} = 'static';
+}
+
+sub specific_werror {
+    my ($config) = @_;
+    my $restore = _to_probe_dir();
+
+    if ($config->{cc} ne 'gcc') {
+        $config->{can_err_decl_after_stmt} = 1;
+        return;
+    }
+
+    my $file = 'try.c';
+    _spew($file, <<'EOT');
+#include <stdlib.h>
+
+int main(int argc, char **argv) {
+     return EXIT_SUCCESS;
+}
+EOT
+
+    print ::dots('    probing support of -Werror=*');
+
+    (my $obj = $file) =~ s/\.c/$config->{obj}/;
+    my $command = "gcc -Werror=declaration-after-statement $config->{ccout}$obj try.c >$devnull 2>&1";
+    my $can_specific_werror = !( system $command );
+
+    print $can_specific_werror ? "YES\n": "NO\n";
+    $config->{can_specific_werror} = $can_specific_werror || 0
 }
 
 

